src/cli/near.rs

```diff
@@ -232,8 +375,164 @@ pub async fn execute_command<T: AsRef<str>>(
                 println!("{:?}", client.get_bridge_prover().await);
             }
+            ReadCommand::GetChainId => {
+                let chain_id = {
+                    let result = client.near_view_call("get_chain_id".into(), vec![]).await?;
+                    U256::from_big_endian(&result.result).low_u64()
+                };
+                println!("{:?}", chain_id);
+            }
+            ReadCommand::GetUpgradeIndex => {
+                let upgrade_index = {
+                    let result = client
+                        .near_view_call("get_upgrade_index".into(), vec![])
+                        .await?;
+                    U256::from_big_endian(&result.result).low_u64()
+                };
+                println!("{:?}", upgrade_index);
+            }
+            ReadCommand::GetBlockHash => {
+                let block_hash = {
+                    let result = client
+                        .near_view_call("get_block_hash".into(), vec![])
+                        .await?;
+                    U256::from_big_endian(&result.result).low_u64()
+                };
+                println!("{:?}", block_hash);
+            }
+            ReadCommand::GetCode { address_hex } => {
+                let code = client
+                    .near_view_call("get_code".into(), address_hex.as_bytes().to_vec())
+                    .await?
+                    .result;
+                println!("{:?}", code);
+            }
+            ReadCommand::GetBalance { address_hex } => {
+                let balance = {
+                    let result = client
+                        .near_view_call("get_balance".into(), address_hex.as_bytes().to_vec())
+                        .await?;
+                    U256::from_big_endian(&result.result).low_u64()
+                };
+                println!("{:?}", balance);
+            }
+            ReadCommand::GetNonce { address_hex } => {
+                let nonce = {
+                    let result = client
+                        .near_view_call("get_nonce".into(), address_hex.as_bytes().to_vec())
+                        .await?;
+                    U256::from_big_endian(&result.result).low_u64()
+                };
+                println!("{:?}", nonce);
+            }
+            ReadCommand::GetStorageAt {
+                address_hex,
+                key_hex,
+            } => {
+                let mut buffer: Vec<u8> = Vec::new();
+                let input = GetStorageAtInput {
+                    address: Address::decode(&address_hex).unwrap(),
+                    key: hex::decode(key_hex).unwrap(),
+                };
+                input.serialize(&mut buffer)?;
+                let storage = {
+                    let result = client
+                        .near_view_call("get_storage_at".into(), buffer)
+                        .await?;
+                    U256::from_big_endian(&result.result).low_u64()
```

Comment:
It probably makes more sense to hex-encode the output instead of turning it into a number. Ethereum storage can contain arbitrary bytes which may or may not semantically represent a number.

Reply:
ok